scripts/chroot.sh

```diff
@@ -30,33 +30,33 @@ if [ -n "${KEY}" ]
 then
     echo "Adding APT key: ${KEY}"
     apt-key add "${KEY}"
+    echo $KEY
 fi
 
 # Add all distro PPAs
-if [ $# -gt 0 ]
+if [ -n "${LIVEBRANCHES}" ] && [ -n "${CODENAME}" ]
 then
     echo "Enabling repository source"
-    ENABLE_SOURCE=--enable-source
-    for repo in "$@"
-    do
-        if [ "$repo" == "--" ]
-        then
-            echo "Disabling repository source"
-            ENABLE_SOURCE=
-        else
-            echo "Adding repository '$repo'"
-            if [[ "${repo}" == "deb "* ]]
-            then
-                echo "${repo}" >> /etc/apt/sources.list
-                if [ -n "${ENABLE_SOURCE}" ]
-                then
-                    echo "${repo}" | sed 's/^deb /deb-src /' >> /etc/apt/sources.list
-                fi
-            else
-                add-apt-repository ${ENABLE_SOURCE} --yes "${repo}"
-            fi
-        fi
+		for newrepo in $LIVEBRANCHES
+		do
+		    echo "Adding preference for '$newrepo'"
+			  tee "/etc/apt/preferences.d/pop-os-staging-${newrepo//./_}" > /dev/null <<-EOF
+Package: *
+Pin: release o=pop-os-staging-$newrepo
+Pin-Priority: 1002
+EOF
+
+			  echo "Adding repository for '$newrepo'"
+			  LINE="deb http://apt.pop-os.org/staging/$newrepo ${CODENAME} main"
+			  if command -v apt-manage
+			  then
+				    apt-manage add "popdev:${LINE}" -t
```

Comment:
This should be `"popdev:${newrepo}"` rather than `"popdev:${LINE}"`.

Reply:
a420bff855d8a6256c1b4cc34262b6af2f5ba875